diff -uNr a/main/exteplayer.c b/main/exteplayer.c
--- a/main/exteplayer.c	2022-06-22 23:26:10.000000000 +0200
+++ b/main/exteplayer.c	2022-06-23 07:38:12.162981700 +0200
@@ -207,6 +207,7 @@
 
 static void map_inter_file_path(char *filename)
 {
+    char *param = NULL;
     if (0 == strncmp(filename, "iptv://", 7))
     {
         FILE *f = fopen(filename + 7, "r");
@@ -224,6 +225,55 @@
             }
         }
     }
+
+    param = strstr(filename, ".mpd&");
+    if (param)
+    {
+        int pos = param - filename;
+        param+=5;
+        int len = strlen(param)+1;
+        int idx = 0;
+        int start = -1;
+        int end = -1;
+        while(idx < len)
+        {
+            if(param[idx] == '=')
+                start = idx-1;
+        
+            if((param[idx] == '\0') || (param[idx] == '\n') || (param[idx] == '&'))
+                end = idx;
+ 
+            if(start != -1 && end != -1)
+            {
+                char str[128];
+                strncpy(str, param+start, end-start);
+                str[end-start] = '\0';
+
+                switch(str[0])
+                {
+                case '0':
+                    ffmpeg_av_dict_set("video_rep_index", str+2, 0);
+                    break;
+                case '1':
+                    ffmpeg_av_dict_set("audio_rep_index", str+2, 0);
+                    break;
+                case '5':
+                    ffmpeg_av_dict_set("cenc_decryption_key", str+2, 0);
+                    break;
+                case '6':
+                    ffmpeg_av_dict_set("cenc_decryption_video_key", str+2, 0);
+                    break;
+                case '7':
+                    ffmpeg_av_dict_set("cenc_decryption_audio_key", str+2, 0);
+                    break;
+                }
+                start = end = -1;
+            }
+            idx++;
+        }
+        filename[pos+4] = '\0';
+    }
+
 }
 
 static int kbhit(void)
@@ -424,7 +474,7 @@
     int digit_optind = 0;
     int aopt = 0, bopt = 0;
     char *copt = 0, *dopt = 0;
-    while ( (c = getopt(argc, argv, "G:W:H:A:V:U:we3dlsrimva:n:x:u:c:h:o:p:P:t:9:0:1:4:f:b:F:S:O:")) != -1) 
+    while ( (c = getopt(argc, argv, "G:W:H:A:V:U:we3dlsrimva:n:x:u:c:h:o:p:P:t:9:0:1:5:6:7:4:f:b:F:S:O:")) != -1) 
     {
         switch (c) 
         {
@@ -547,6 +597,15 @@
             break;
         case '1':
             ffmpeg_av_dict_set("audio_rep_index", optarg, 0);
+            break;            
+        case '5':
+            ffmpeg_av_dict_set("cenc_decryption_key", optarg, 0);
+            break;
+        case '6':
+            ffmpeg_av_dict_set("cenc_decryption_video_key", optarg, 0);
+            break;
+        case '7':
+            ffmpeg_av_dict_set("cenc_decryption_audio_key", optarg, 0);
             break;
         case '4':
 #ifdef HAVE_FLV2MPEG4_CONVERTER
@@ -663,8 +722,11 @@
         printf("[-u user-agent] set custom http User-Agent header\n");
         printf("[-c cookies] set cookies - not working at now, please use -h instead\n");
         printf("[-x separateAudioUri]\n");
-        printf("[-0 idx] video MPEG-DASH representation index\n");
-        printf("[-1 idx] audio MPEG-DASH representation index\n");
+        printf("[-0 idx] video MPEG-DASH stream index\n");
+        printf("[-1 idx] audio MPEG-DASH stream index\n");
+        printf("[-5 idx] all stream MPEG-DASH cenc decryption key\n");
+        printf("[-6 idx] video MPEG-DASH stream index cenc decryption key\n");
+        printf("[-7 idx] audio MPEG-DASH stream index cenc decryption key\n");
         printf("[-f ffopt=ffval] any other ffmpeg option\n");
         printf("[-F path to additional file with moov atom data (used for mp4 playback in progressive download mode)\n");
         printf("[-O moov atom offset in the original file (used for mp4 playback in progressive download mode)\n");

